site/content/docs/custom-layout.md

```diff
@@ -0,0 +1,149 @@
+---
+title: How to create custom page types and layouts
+description: You may have many different types of notes that you want to display in different ways on your Flowershow website. In this tutorial we will cover how to add a new `Recipe` document type for our cooking blog as well as a custom `RecipeLayout` for that specific document type.
+type: Blog
+created: 2023-03-02
+authors: [philippe-du-preez]
+---
+
+You may have many different types of notes that you want to display in different ways on your Flowershow website. In this tutorial we will cover how to add a new `Recipe` document type for our cooking blog as well as a custom `RecipeLayout` for that specific document type. Then we will create a home page for listing all the pages of this type.
+
+Let's start with creating a new `Recipe` document type.
+
+## Create a custom Contentlayer document type
+
+In order to create our new `Recipe` document type, we need to go inside `.flowershow/contentlayer.config.ts` and add instructions to tell Contentlayer how to parse:
+
+- `name`: namespace
+- `filePathPattern`: input files (Files within this path do not need to specify their `Type` to `Recipe`)
+- `fields`: meta data fields
+- `computedFields`: derived meta data fields
+
+```javascript
+//contentlayer.config.ts
+// ...
+const recipeFields = {
+  created: { type: "date", required: true },
+  authors: {
+    type: "list",
+    of: { type: "string" },
+  },
+  layout: { type: "string", default: "recipe" },
+};
+
+const Recipe = defineDocumentType(() => ({
+  name: "Recipe",
+  filePathPattern: "recipes/!(index)*.md*",
+  contentType: "mdx",
+  fields: {
+    ...sharedFields,
+    ...recipeFields,
+  },
+  computedFields,
+}));
+
+export default makeSource({
+  // ...
+  documentTypes: [Recipe, Page],
+  // ...
+});
+```
+
+We can reuse the sharedFields (title, description etc.) as needed and add some custom fields, `recipeFields`, for our `Recipe` type. These fields will be accessible in the `RecipeLayout` we will be making. We do not have to include the `Type`, since we are adding the file in the `content/recipes` directory. We can then add the needed fields to the frontmatter of our recipe posts, like so:
+
+```bash
+---
+title: Homemade Lasagne Recipe
+description: Lasagne is a classic Italian dish that is perfect for any occasion. It is a hearty and delicious meal that can be enjoyed by everyone.
+created: 2022-11-29
+authors: [John Doe]
+---
+Lasagne is a classic Italian dish that is perfect for any occasion. It is a hearty and delicious meal ...
+```
+
+>See more on how to use the `Author` document type [here](https://flowershow.app/docs/blog#blog-authors)
```

Comment:
You can use callouts for extra info like that. Take a look at our docs on callouts/admonitions.